gamemodes/terrortown/entities/entities/ttt_base_placeable.lua

```diff
@@ -0,0 +1,298 @@
+
+			if vecHitNormal then
+				self:SetAngles(vecHitNormal:Angle() + (self:GetStickRotation() or Angle(0, 0, 0)))
+			end
+
+			local pos = self:GetPos()
+			local ignore = player.GetAll()
+
+			ignore[#ignore + 1] = {self}
+
+			local tr = util.TraceEntity({
+				start = pos,
+				endpos = pos - Vector(0, 0, 16),
+				filter = ignore,
+				mask = MASK_SOLID
+			}, self)
+
+			sound.Play(soundWeld, pos, 75)
+
+			if tr.Hit and (IsValid(tr.Entity) or tr.HitWorld) then
+				local phys = self:GetPhysicsObject()
+
+				if IsValid(phys) then
+					if tr.HitWorld then
+						phys:EnableMotion(false)
+					else
+						self.originalMass = phys:GetMass()
+						phys:SetMass(150)
+					end
+				end
+
+				-- only weld to objects we cannot pick up
+				local entphys = tr.Entity:GetPhysicsObject()
+				if IsValid(entphys) and entphys:GetMass() > CARRY_WEIGHT_LIMIT then
+					constraint.Weld(self, tr.Entity, 0, 0, 0, true)
+				end
+			end
+		else
+			constraint.RemoveConstraints(self, "Weld")
+
+			local phys = self:GetPhysicsObject()
+
+			if IsValid(phys) then
+				phys:EnableMotion(true)
+				phys:SetMass(self.originalMass or 10)
+			end
+		end
+	end
+
+	---
+	-- Checks if the entity is welded to a surface.
+	-- @return boolean Returns true if the entity is welded to a surface
+	-- @realm server
+	function ENT:IsWeldedToSurface()
+		return self.stateWelding or false
+	end
+
+	---
+	-- Helper function for a weapon that wants to throw the entity. Already handles everything.
+	-- @param Player ply The player that throws the entity, the owner
+	-- @param[opt] Angle rotationalOffset The model's rotational offset that should be applied
+	-- @return boolean Returns true on success
+	-- @realm server
+	function ENT:ThrowEntity(ply, rotationalOffset)
+		ply:SetAnimation(PLAYER_ATTACK1)
+
+		rotationalOffset = rotationalOffset or Angle(0, 0, 0)
+
+		local posThrow = ply:GetShootPos() - Vector(0, 0, 15)
+		local vecAim = ply:GetAimVector()
+		local velocity = ply:GetVelocity()
+		local velocityThrow = velocity + vecAim * 250
+
+		self:SetPos(posThrow + vecAim * 10)
+		self:SetOriginator(ply)
+		self:Spawn()
+		self:PointAtEntity(ply)
+
+		local ang = self:GetAngles()
+		ang:RotateAroundAxis(ang:Right(), rotationalOffset.pitch)
+		ang:RotateAroundAxis(ang:Up(), rotationalOffset.yaw)
+		ang:RotateAroundAxis(ang:Forward(), rotationalOffset.roll)
+
+		self:SetAngles(ang)
+		self:PhysWake()
+
+		local phys = self:GetPhysicsObject()
+
+		if IsValid(phys) then
+			phys:SetVelocity(velocityThrow)
+		end
+
+		self:EmitSound(soundThrow)
+
+		return true
+	end
+
+	---
+	-- Helper function for a weapon that wants to stick the entity to a surface. Already handles everything.
+	-- @param Player ply The player that sticks the entity, the owner
+	-- @param[opt] Angle rotationalOffset The model's rotational offset that should be applied
+	-- @param[opt] number angleCondition The angle condition that has to be met to apoply the rotational offset
```

Comment:
These two are very hard to understand, we should explain them more

Also: typo in apply